src/components/ModalPayment/ModalMegaGam/ModalMegaGam.tsx

```diff
@@ -0,0 +1,373 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import styles from './styles.module.scss';
+import {
+  useModalPaymentDispatch,
+  ModalPaymentConstants,
+  ModalPaymentTypes,
+  useModalPaymentState,
+} from '../../../utilities/hooks/ModalPaymentContext';
+import { useTranslation, Trans } from 'react-i18next';
+import ReactPixel from 'react-facebook-pixel';
+import CampaignProgressBar from '../../MerchantsPage/gam/CampaignProgressBar';
+import { SIZE_TYPE } from '../../MerchantsPage/gam/ProgressBar';
+import styled from 'styled-components';
+import { formatCurrency } from '../../../utilities/general/textFormatter';
+import { calculateFeeAmount } from '../../../utilities/general/feeCalculator';
+import FiscalSponsor from '../../MerchantsPage/gam/FiscalSponsor';
+import { Tooltip } from '@material-ui/core';
+import { Help } from '@material-ui/icons';
+import { COST_LIMIT_DOLLARS } from '../../../consts';
+import { phoneScreens } from '../../../utilities/general/responsive';
+import { useEffect } from 'react';
+
+export interface Props {
+  sellerId: string;
+  sellerName: string;
+  costPerMealInDollars: number;
+  nonProfitLocationId?: string;
+}
+
+export const Modal = (props: Props) => {
+  const { t } = useTranslation();
+  const dispatch = useModalPaymentDispatch(null);
+
+  // set initial number of meals to 1
+  const { fees, campaignState } = useModalPaymentState(null);
+  const [numberOfMeals, setNumberOfMeals] = useState(1);
+
+  // TODO (billy-yuan): Confirm whether mega-gam campaigns in the future will have donation matching.
+  // If so, we need to add a flag such as isMatching in the back end to determine whether to calculate a matchAmount
+
+  const calculateMatchAmount = (userDonationDollars: number) => {
+    if (
+      !campaignState &&
+      !campaignState.target_amount
```

Comment:
I'm pretty sure that this is required in the model

Reply:
Unless you wanna be extra safe on the frontend

Reply:
got it, i can remove. once we add in `match_amount` in the back end, we can remove this entire if statement and just pass in the `match_amount` directly from the api